packages/netlify-cms-backend-bitbucket/src/implementation.js

```diff
@@ -225,36 +227,70 @@ export default class BitbucketBackend {
     }));
   }
 
-  getMedia() {
+  getMedia(mediaFolder = this.config.get('media_folder')) {
     return this.api
-      .listAllFiles(this.config.get('media_folder'))
+      .listAllFiles(mediaFolder)
       .then(files =>
         files.map(({ id, name, path }) => ({ id, name, path, displayURL: { id, path } })),
       );
   }
 
+  getMediaAsBlob(path, id) {
+    return this.api.readFile(path, id, { parseText: false });
+  }
+
   getMediaDisplayURL(displayURL) {
     this._mediaDisplayURLSem = this._mediaDisplayURLSem || semaphore(MAX_CONCURRENT_DOWNLOADS);
     const { id, path } = displayURL;
     return new Promise((resolve, reject) =>
       this._mediaDisplayURLSem.take(() =>
-        this.api
-          .readFile(path, id, { parseText: false })
+        this.getMediaAsBlob(path, id)
           .then(blob => URL.createObjectURL(blob))
           .then(resolve, reject)
           .finally(() => this._mediaDisplayURLSem.leave()),
       ),
     );
   }
 
+  async getMediaFile(path) {
+    const name = basename(path);
+    const blob = await this.getMediaAsBlob(path, null);
+    const fileObj = new File([blob], name);
+    const url = URL.createObjectURL(fileObj);
+
+    return {
+      displayURL: url,
+      path,
+      name,
+      size: fileObj.size,
+      file: fileObj,
+      url,
+    };
+  }
+
   persistEntry(entry, mediaFiles, options = {}) {
     return this.api.persistFiles([entry], options);
   }
 
   async persistMedia(mediaFile, options = {}) {
-    await this.api.persistFiles([mediaFile], options);
-    const { value, path, fileObj } = mediaFile;
-    return { name: value, size: fileObj.size, path: trimStart(path, '/k') };
+    const { fileObj } = mediaFile;
+
+    const [sha] = await Promise.all([
+      getBlobSHA(fileObj),
+      this.api.persistFiles([mediaFile], options),
+    ]);
+
+    const url = URL.createObjectURL(fileObj);
+
+    return {
```

Comment:
Until now the `persistMedia` action filled in the blanks for missing the implementation in the backneds.
I think it makes sense that all backends return the same object structure.